### PR TITLE
Migrate old tests to colocation

### PR DESCRIPTION
**Pull request by MartinquaXD**
# Description
Eventually we want to get rid of the `solver` crate. To do that we must migrate all the e2e tests to no longer use the old driver.

# Changes
Converts all e2e tests that used the old architecture to use colocated baseline, driver and autopilot.
The conversion is mostly straight forward. Some tests required very slight adjustments to asserted values.

## How to test
This is the test

## TODO before merge
- 1 regular e2e test is still failing
- 1 fork based test still relies on the old infra
- refactor some of the changes and clean up in the end
- investigate reason I had to adjust some asserted values by tiny amounts

### CODE REVIEW

**Comment by sunce86** [comment]
### How to test
This is the test

😄

**Comment by sunce86** [comment]
I propose to merge this PR as it's blocking the merge of https://github.com/cowprotocol/services/pull/2190
I tried fixing the remaining issues, but couldn't figure out why differences. 

Although not finished, it's definitely a step forward (right now old e2e tests check new code changes against old driver which is useless). I can create a new issue to address the remaining fixes.

@fleupold wdyt

File: crates/e2e/tests/e2e/eth_integration.rs [hunk 1]
```diff
@@ -128,6 +125,6 @@ async fn eth_integration(web3: Web3) {
     );
     assert_eq!(
         trader_b_eth_balance_after - trader_b_eth_balance_before,
-        49_800_747_827_208_136_744_u128.into()
+        49_776_284_258_574_379_601_u128.into()
```
Comment (fleupold) [inline crates/e2e/tests/e2e/eth_integration.rs:128]: This is a difference of 5bips. Do we have any idea where this is coming from?

File: crates/e2e/tests/e2e/onchain_settlement.rs [hunk 2]
```diff
@@ -78,8 +81,21 @@ async fn onchain_settlement(web3: Web3) {
     );
 
     let services = Services::new(onchain.contracts()).await;
-    services.start_autopilot(vec![]);
-    services.start_api(vec![]).await;
+    let solver_endpoint = colocation::start_solver(onchain.contracts().weth.address()).await;
+    colocation::start_driver(
+        onchain.contracts(),
+        vec![SolverEngine {
```
Comment (fleupold) [inline crates/e2e/tests/e2e/onchain_settlement.rs:87]: Maybe we need to start the naive solver here for it to pick up the CoW.

File: crates/e2e/tests/e2e/onchain_settlement.rs [hunk 3]
```diff
@@ -101,7 +117,7 @@ async fn onchain_settlement(web3: Web3) {
     let order_b = OrderCreation {
         sell_token: token_b.address(),
         sell_amount: to_wei(50),
-        fee_amount: to_wei(1),
+        fee_amount: 500_000_000_000_000_000u128.into(),
```
Comment (fleupold) [inline crates/e2e/tests/e2e/onchain_settlement.rs:120]: Why is this change needed?

**Review by fleupold — COMMENTED** [review]
> I tried fixing the remaining issues, but couldn't figure out why differences.

Can we at least fix or adjust the failing test to work? This shouldn't be too much work and would ensure we are not leaving the test base in a bad state.


Also, can we remove the `colocation_*` prefix from the test case files?

File: crates/e2e/tests/e2e/onchain_settlement.rs (hunk 2 above)
Comment (squadgazzz) [reply crates/e2e/tests/e2e/onchain_settlement.rs:87]: That helped, thanks 👍

File: crates/e2e/tests/e2e/onchain_settlement.rs (hunk 3 above)
Comment (squadgazzz) [reply crates/e2e/tests/e2e/onchain_settlement.rs]: No need anymore, reverted

File: crates/e2e/tests/e2e/eth_integration.rs (hunk 1 above)
Comment (squadgazzz) [reply crates/e2e/tests/e2e/eth_integration.rs:128]: With `naive` solver it get's back to the previous value

**Comment by squadgazzz** [comment]
> Also, can we remove the colocation_* prefix from the test case files?
https://github.com/cowprotocol/services/pull/2247

**Review by sunce86 — APPROVED** [review]
🎉
